### PR TITLE
Add "cs" and "python" categories to the `.quiz` command (and more in desc)

### PR DESCRIPTION
**Pull request by Objectivitix**
## Relevant Issues
Closes issues #753 and #747, extends PR #726

## Description
It was planned to incorporate these two categories in the same PR, however I didn't get enough quiz entries for them. But now, thanks to lak, I have enough! So in total, 60 new questions are added. I also changed the starting embed a bit to make it look nicer, made the variation tolerance stricter, and added some missing answers specified in 747 to the math & science questions.

## Did you:
- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

**Comment by Objectivitix** [comment]
lint works, everything works, ready for review :D

**Review by ToxicKidz — CHANGES_REQUESTED** [review]
Hey, you should test before you push, because I get this error when running the code:
```py 
Traceback (most recent call last):
   File "/usr/local/lib/python3.9/site-packages/discord/ext/commands/bot.py", line 618, in _load_from_module_spec
     setup(self)
   File "/bot/bot/exts/evergreen/trivia_quiz.py", line 593, in setup
     bot.add_cog(TriviaQuiz(bot))
   File "/bot/bot/exts/evergreen/trivia_quiz.py", line 202, in __init__
     self.questions = self.load_questions()
   File "/bot/bot/exts/evergreen/trivia_quiz.py", line 222, in load_questions
     return json.loads(p.read_text(encoding="utf-8"))
   File "/usr/local/lib/python3.9/json/__init__.py", line 346, in loads
     return _default_decoder.decode(s)
   File "/usr/local/lib/python3.9/json/decoder.py", line 340, in decode
     raise JSONDecodeError("Extra data", s, end)
 json.decoder.JSONDecodeError: Extra data: line 2 column 10 (char 10)
```

**Comment by Objectivitix** [comment]
@ToxicKidz I did test all 60 questions before, the error arose during the commit (copy paste error smh). It has been resolved :D

File: bot/resources/evergreen/trivia_quiz.json [hunk 1]
```diff
@@ -604,5 +604,309 @@
...
+      "answer": "cell_contents"
+    },
+    {
+      "id": 507,
+      "question": "What name does a lambda function have?",
+      "answer": "<lambda>"
+    },
+    {
+      "id": 508,
+      "question": "What folder contains all special site builtins, such as help or credits?",
```
Comment (ToxicKidz) [inline bot/resources/evergreen/trivia_quiz.json:798]: Isn't [`_sitebuiltins`](https://github.com/python/cpython/blob/main/Lib/_sitebuiltins.py) as file, not a folder?

**Review by ToxicKidz — CHANGES_REQUESTED** [review]

File: bot/resources/evergreen/trivia_quiz.json [hunk 2]
```diff
@@ -604,5 +604,309 @@
...
+  ],
+  "cs": [
+    {
+      "id": 401,
+      "question": "What does HTML stand for?",
+      "answer": "Hyper Text Markup Language"
+    },
+    {
+      "id": 402,
+      "question": "What's ASCII's full name?",
```
[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json]: got it, resolved

File: bot/resources/evergreen/trivia_quiz.json [hunk 3]
```diff
@@ -604,5 +604,309 @@
...
+    },
+    {
+      "id": 528,
+      "question": "What module is the function to set a function that runs for every line of python code present in?",
+      "answer": "sys"
+    },
+    {
+      "id": 529,
+      "question": "Which of these is an `Exception`? `NotImplemented`, `asyncio.CancelledError`, `StopIteration`",
+      "answer": "StopIteration"
```
[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:904]: hm... lak made these questions, i'll go confirm with him

[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:904]: from python docs:
> Changed in version 3.8: CancelledError is now a subclass of BaseException.

File: bot/resources/evergreen/trivia_quiz.json (hunk 1 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:798]: will change, resolved :D

File: bot/resources/evergreen/trivia_quiz.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:904]: i will wait for lak's response before further action

[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:904]: from lak:
> It is a BaseException, not an Exception. Though maybe it should be rephrased to clarify meaning
i will rephrase it to be more precise and less vague

[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json]: resolved :D

File: bot/resources/evergreen/trivia_quiz.json [hunk 4]
```diff
@@ -372,7 +372,7 @@
     {
       "id": 216,
       "question": "In set builder notation, {p/q | q ≠ 0, p & q ∈ Z} represents what?",
-      "answer": "Rational Numbers"
+      "answer": "rationals, Rational Numbers"
```
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:375]: Maybe capitalise rationals to keep consistency.

File: bot/resources/evergreen/trivia_quiz.json [hunk 5]
```diff
@@ -604,5 +604,309 @@
       "question": "When DNA is unzipped, two strands are formed. What are they called (separate both answers by the word \"and\")?",
       "answer": "leading and lagging, leading strand and lagging strand"
     }
+  ],
+  "cs": [
+    {
+      "id": 401,
+      "question": "What does HTML stand for?",
+      "answer": "Hyper Text Markup Language"
```
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:612]: Hyper Text should not be separated by a space.

File: bot/resources/evergreen/trivia_quiz.json [hunk 6]
```diff
@@ -604,5 +604,309 @@
...
+    },
+    {
+      "id": 412,
+      "question": "How many bits are in a TCP checksum header?",
+      "answer": "16, sixteen"
+    },
+    {
+      "id": 413,
+      "question": "Which protocol handles communication between email servers?",
+      "answer": "SMTP"
```
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:672]: This could be changed to "Give one protocol that handles mail" as there are SMTP, IMAP and POP3 that are used today

File: bot/resources/evergreen/trivia_quiz.json [hunk 7]
```diff
@@ -604,5 +604,309 @@
...
+      "question": "Which of these languages is the youngest? Lisp, Python, Java, Haskell, Prolog, Ruby, Perl",
+      "answer": "Java"
+    }
+  ],
+  "python": [
+    {
+      "id": 501,
+      "question": "Is it all objects (y/n)?",
+      "answer": "y, yes"
+    },
```
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:765]: Is this really necessary?

File: bot/resources/evergreen/trivia_quiz.json [hunk 8]
```diff
@@ -604,5 +604,309 @@
...
+      "answer": "@"
+    },
+    {
+      "id": 516,
+      "question": "Which standard library module is designed for making simple parsers for languages like shell, as well as safe quoting of strings for use in a shell?",
+      "answer": "shlex"
+    },
+    {
+      "id": 517,
+      "question": "Which one of these protocols/abstract base classes does the builtin `range` object NOT implement? `Sequence`, `Iterable`, `Generator`",
```
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:843]: Maybe keep consistency here and enclose multiple choice options in parentheses or do the inverse

File: bot/resources/evergreen/trivia_quiz.json [hunk 9]
```diff
@@ -604,5 +604,309 @@
...
+    },
+    {
+      "id": 524,
+      "question": "Which protocol is used to pass self to methods when accessed on classes?",
+      "answer": "Descriptor"
+    },
+    {
+      "id": 525,
+      "question": "What class was added in a new standard library module for python 3.9?",
+      "answer": "TopologicalSorter"
```
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:884]: there was more than one module added to the stdlib in python 3.9 iirc.

File: bot/resources/evergreen/trivia_quiz.json [hunk 10]
```diff
@@ -604,5 +604,309 @@
...
+    },
+    {
+      "id": 516,
+      "question": "Which standard library module is designed for making simple parsers for languages like shell, as well as safe quoting of strings for use in a shell?",
+      "answer": "shlex"
+    },
+    {
+      "id": 517,
+      "question": "Which one of these protocols/abstract base classes does the builtin `range` object NOT implement? `Sequence`, `Iterable`, `Generator`",
+      "answer": "Generator"
```
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:844]: Maybe keep consistency by enclosing multiple choice options in parentheses.

File: bot/resources/evergreen/trivia_quiz.json [hunk 11]
```diff
@@ -604,5 +604,309 @@
...
+    },
+    {
+      "id": 422,
+      "question": "Which of these languages is the oldest? Tcl, Smalltalk 80, haskell, standard ML, Java",
+      "answer": "Smalltalk 80"
+    },
+    {
+      "id": 423,
+      "question": "What is the name for unicode codepoints that do not fit into 16 bits",
+      "answer": "surrogates"
```
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:722]: Make sure that you add question marks to the end of questions.

File: bot/resources/evergreen/trivia_quiz.json (hunk 6 above)
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:672]: Maybe reword this to "What is the most popular email protocol?" as IMAP and POP3 are still in use.

File: bot/resources/evergreen/trivia_quiz.json (hunk 4 above)
Comment (Kronifer) [inline bot/resources/evergreen/trivia_quiz.json:375]: Maybe capitalise this for consistency, check for capitalisation in other areas as ive noticed this before.

**Review by Kronifer — CHANGES_REQUESTED** [review]
Few nitpicks here

**Comment by Kronifer** [comment]
Github seems to have added duplicate change requests, i did my best to remove them.

File: bot/resources/evergreen/trivia_quiz.json (hunk 4 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:375]: sure, will be resolved in the next commit

File: bot/resources/evergreen/trivia_quiz.json [hunk 12]
```diff
@@ -604,5 +604,309 @@
...
+      "answer": "D"
+    },
+    {
+      "id": 421,
+      "question": "What is a HTTP/1.1 feature that was superseded by HTTP/2 multiplexing and is unsupported in most browsers nowadays?",
+      "answer": "pipelining"
+    },
+    {
+      "id": 422,
+      "question": "Which of these languages is the oldest? Tcl, Smalltalk 80, haskell, standard ML, Java",
```
[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:716]: got it

File: bot/resources/evergreen/trivia_quiz.json (hunk 7 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:765]: yes.

File: bot/resources/evergreen/trivia_quiz.json (hunk 8 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:843]: there is consistency - none of these non-lettered ones are enclosed in parentheses

File: bot/resources/evergreen/trivia_quiz.json (hunk 9 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:884]: hm, i'll check with lak

File: bot/resources/evergreen/trivia_quiz.json (hunk 6 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:672]: ok

File: bot/resources/evergreen/trivia_quiz.json (hunk 9 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:884]: lak never responded- i'll think up a new question and replace it

Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json]: resolved

**Comment by Objectivitix** [comment]
@Kronifer @ToxicKidz all your changes have been implemented.

File: bot/resources/evergreen/trivia_quiz.json [hunk 13]
```diff
@@ -604,5 +604,309 @@
...
+    {
+      "id": 430,
+      "question": "Which of these languages is the youngest? Lisp, Python, Java, Haskell, Prolog, Ruby, Perl",
+      "answer": "Java"
+    }
+  ],
+  "python": [
+    {
+      "id": 501,
+      "question": "Is it all objects (y/n)?",
```
Comment (vcokltfre) [inline bot/resources/evergreen/trivia_quiz.json:763]: I see that there's a comment marked as resolved here that asks if it's necessary, which was just answered with 'yes'. While I'm personally unsure whether this should be a question, at the very least I think it's badly phrased, and will lead to confusion, a more suitable question I feel would be along the lines of "Is everything in Python an object?", since "Is it all objects?" sounds too vague.

File: bot/resources/evergreen/trivia_quiz.json [hunk 14]
```diff
@@ -604,5 +604,309 @@
...
+      "answer": "indices"
+    },
+    {
+      "id": 503,
+      "question": "What exception, other than `StopIteration`, can you raise from a `__getitem__` dunder to indicate to an iterator that it should stop?",
+      "answer": "IndexError"
+    },
+    {
+      "id": 504,
+      "question": "What type does the `&` operator return given 2 `dict_keys` objects?",
```
Comment (vcokltfre) [inline bot/resources/evergreen/trivia_quiz.json:778]: This question tripped me up when first reading it because it seems a little ambiguous at first, I think "return when given 2..." would make it more understandable at a glance

**Review by vcokltfre — CHANGES_REQUESTED** [review]

File: bot/resources/evergreen/trivia_quiz.json [hunk 15]
```diff
@@ -604,5 +604,309 @@
...
+    },
+    {
+      "id": 412,
+      "question": "How many bits are in a TCP checksum header?",
+      "answer": "16, sixteen"
+    },
+    {
+      "id": 413,
+      "question": "What is the most popular protocol (as of 2021) that handles communication between email servers?",
+      "answer": "SMTP"
```
[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:672]: ah right, to be consistent with "MX, mail exchange" as well.

File: bot/resources/evergreen/trivia_quiz.json [hunk 16]
```diff
@@ -604,5 +604,309 @@
...
+      "answer": "2038"
+    },
+    {
+      "id": 408,
+      "question": "What are the components of digital devices that make up logic gates called?",
+      "answer": "transistors"
+    },
+    {
+      "id": 409,
+      "question": "How many possible public IPv6 addresses are there (answer in 2^n)?",
```
[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:651]: got it

File: bot/resources/evergreen/trivia_quiz.json [hunk 17]
```diff
@@ -604,5 +604,309 @@
...
+      "answer": "NAND, NOR"
+    },
+    {
+      "id": 419,
+      "question": "What is the mathematical formalism which functional programming was built on?",
+      "answer": "lambda calculus"
+    },
+    {
+      "id": 420,
+      "question": "Why is a DDoS attack different from a DoS attack (answer in letters - A. because the victim's server was indefinitely disrupted from the amount of traffic, B. because it also attacks the victim's confidentiality, C. because the attack had political purposes behind it, D. because the traffic flooding the victim originated from many different sources)?",
```
[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:706]: yeah, i agree. it will be refactored

File: bot/resources/evergreen/trivia_quiz.json [hunk 18]
```diff
@@ -604,5 +604,309 @@
...
+    },
+    {
+      "id": 424,
+      "question": "Under what locale does making a string lowercase behave differently?",
+      "answer": "Turkish"
+    },
+    {
+      "id": 425,
+      "question": "What does the \"a\" represent in a HSLA color value?",
+      "answer": "transparency, translucency, alpha value, alpha channel"
```
[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:732]: got it

File: bot/resources/evergreen/trivia_quiz.json (hunk 13 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:763]: i agree with the last part, the question should be phrased better.

File: bot/resources/evergreen/trivia_quiz.json (hunk 14 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:778]: right, you gotta add 'em conjunctions

**Comment by Objectivitix** [comment]
@vcokltfre The changes will be implemented later, currently I gotta update my website lol :D

File: bot/resources/evergreen/trivia_quiz.json (hunk 16 above)
[the remark this answers is not in the document]
Comment (vcokltfre) [reply bot/resources/evergreen/trivia_quiz.json:651]: This isn't resolved

[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:651]: WAT- i thought i deleted the word lol

[the remark this answers is not in the document]
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json]: got it, resolved :D

**Review by vcokltfre — APPROVED** [review]
LGTM! :shipit:

File: bot/resources/evergreen/trivia_quiz.json [hunk 19]
```diff
@@ -604,5 +604,309 @@
...
+      "answer": "American Standard Code for Information Interchange"
+    },
+    {
+      "id": 403,
+      "question": "What does SASS stand for?",
+      "answer": "Syntactically Awesome Stylesheets, Syntactically Awesome Style Sheets"
+    },
+    {
+      "id": 404,
+      "question": "In neural networks, _______________ is an algorithm for supervised learning using gradient descent.",
```
Comment (ToxicKidz) [inline bot/resources/evergreen/trivia_quiz.json:626]: Should these underscores be escaped?

**Review by ToxicKidz — CHANGES_REQUESTED** [review]

File: bot/resources/evergreen/trivia_quiz.json [hunk 20]
```diff
@@ -604,5 +604,309 @@
...
+      "answer": "next"
+    },
+    {
+      "id": 527,
+      "question": "Is the `__aiter__` method async (y/n)?",
+      "answer": "n, no"
+    },
+    {
+      "id": 528,
+      "question": "What module is the function to set a function that runs for every line of python code present in?",
```
Comment (ToxicKidz) [inline bot/resources/evergreen/trivia_quiz.json:898]: (It seems this comment was not added in the review)
This question doesn't make a lot of sense.

File: bot/resources/evergreen/trivia_quiz.json (hunk 19 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:626]: Oh yeah! I'll do that

File: bot/resources/evergreen/trivia_quiz.json (hunk 20 above)
Comment (Objectivitix) [reply bot/resources/evergreen/trivia_quiz.json:898]: Got it, will replace
~~lak please explain~~

**Review by ToxicKidz — APPROVED** [review]
:shipit:

**Comment by Objectivitix** [comment]
petapoggers :D one last core dev appr then we done

**Review by HassanAbouelela — APPROVED** [review]
